crates/fuel-core/src/state/iterable_view.rs

```diff
@@ -0,0 +1,85 @@
+use fuel_core_storage::{
+    iter::{
+        BoxedIter,
+        IterDirection,
+        IterableStore,
+    },
+    kv_store::{
+        KVItem,
+        KeyValueInspect,
+        StorageColumn,
+        Value,
+    },
+    Result as StorageResult,
+};
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct IterableViewWrapper<Column>(
+    Arc<dyn IterableStore<Column = Column> + Sync + Send>,
+);
+
+impl<Column> IterableViewWrapper<Column> {
+    pub fn into_inner(self) -> Arc<dyn IterableStore<Column = Column> + Sync + Send> {
+        self.0
+    }
+}
+
+impl<Column> std::fmt::Debug for IterableViewWrapper<Column> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("IterableViewWrapper").finish()
+    }
+}
+
+impl<Column> IterableViewWrapper<Column> {
+    pub fn new(storage: Arc<dyn IterableStore<Column = Column> + Sync + Send>) -> Self {
```

Comment:
This concept of passing an `Arc<dyn Trait>` into `new` seems like a leak of the internal details. Same with returning it from `into_inner`. I'd expect `new` to take the inner trait object and wrap the `Arc` for you.

I don't know if I have a solution though. I feel like these helpers exist to solve specific situations, like converting to `KeyValueView` maybe we can just convert directly without exposing the inner `Arc`? I haven't dug into it enough to say and maybe we just need to settle with this.

Reply:
Do we need the `into_inner` to return `Arc`?

Reply:
Yep, since we don't know what is type used inside of the `Arc`=) We can't extract it back.

Reply:
That makes sense.

How do you feel about a

```
impl<C> From<IterableViewWrapper<C>> for Arc<dyn IterableStore<Column = C> + Sync + Send> 
```

Reply:
I prefer to not overload `into` and have explicit `into_inner`. It makes clear that the type acts as a container.

Reply:
Okay. I'm fine with that.